### DISCUSSION TITLE
Adding srxl2 support for as a serial protocol

### DISCUSSION THREAD

**Discussion by snppla** [ExpressLRS/ExpressLRS#2972] [Ideas]
Avian ESC's are able to talk to Spektrum receiver's via the srxl2 protocol. Modern Spektrum receivers automatically detect this protocol and communicate with the ESC via this protocol. Most of Horizon's bnf and pnp airplanes include Avian ESC's. This is how many of their planes are able to have their esc reverse and get ESC related telemetry (volts, amps, temperature, rpm). I think it would be useful to add srxl2 as a serial option to receivers so that on aircraft that use these ESC's, elrs can support the reverse thrust and telemetry features. Spektrum publish's this protocol, so the specification of how to implement this already exist.

**Comment by Majorsass** [comment]
I second this idea.

Not only the AVIAN ESCs support SRXL2, but also all  FIRMA ground based ESCs too. Since the release of ground vehicle ELRS TX/RX, there are many more possible useful applications.

The Spec is available here:

https://github.com/SpektrumRC/SRXL2/blob/master/Docs/SRXL2%20Specification.pdf

**Comment by RobertoD91** [comment]
If can be useful, this project implement Spektrum esc control via SRXL2: https://github.com/dgatf/msrc/
